File: en/java-spring-demo/src/main/java/com/itangcent/springboot/demo/common/model/TestJsonFieldBean.java
package com.itangcent.springboot.demo.common.model;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * JSON Field Name Mapping Test Bean
 * Used to demonstrate Jackson's @JsonProperty annotation functionality
 * Implements custom mapping between Java field names and JSON field names
 */
public class TestJsonFieldBean {
    /**
     * Username
     * Maps to "user-name" during JSON serialization
     */
    @JsonProperty("user-name")
    private String userName;

    /**
     * User Age
     * Maps to "user-age" during JSON serialization
     */
    @JsonProperty("user-age")
    private Integer userAge;

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    public Integer getUserAge() {
        return userAge;
    }

    public void setUserAge(Integer userAge) {
        this.userAge = userAge;
    }
} 

File: cn/kotlin-spring-demo/src/main/kotlin/com/itangcent/common/model/TestJsonFieldBean.kt
package com.itangcent.common.model

import com.fasterxml.jackson.annotation.JsonProperty

/**
 * JSON字段名映射测试Bean
 * 用于演示Jackson的@JsonProperty注解功能
 * 实现Java字段名和JSON字段名之间的自定义映射
 */
data class TestJsonFieldBean(
    /**
     * 用户名
     * 在JSON序列化中映射为"user-name"
     */
    @JsonProperty("user-name")
    var userName: String? = null,

    /**
     * 用户年龄
     * 在JSON序列化中映射为"user-age"
     */
    @JsonProperty("user-age")
    var userAge: Int? = null
) 